<launch>
  <arg name="dataset_name" default="Euroc" />
  <arg name="verbosity" default="0" />
  <arg name="num_robots" default="3" />
  <arg name="mesh_reconstruction" default="true" />
  <arg name="semantic_mesh" default="true" />
  <arg name="environment" default="camp" />
  
  <group ns="kimera0">
    <arg name="robot_id" value="0" />
    <include file="$(find kimera_distributed)/launch/kimera_centralized_agent.launch" pass_all_args="true" />
  </group>

  <group ns="kimera1">
    <arg name="robot_id" value="1" />
    <include file="$(find kimera_distributed)/launch/kimera_centralized_agent.launch" pass_all_args="true" />
  </group>

  <group ns="kimera2">
    <arg name="robot_id" value="2" />
    <include file="$(find kimera_distributed)/launch/kimera_centralized_agent.launch" pass_all_args="true" />
  </group>

  <!-- Launch Basestation -->
  <node name="kimera_centralized_node" pkg="kimera_distributed" 
    type="kimera_centralized_node" output="screen" ns="kimera_centralized">
    <param name="num_robots" value="$(arg num_robots)" />
    <rosparam file="$(find kimera_distributed)/params/centralized.yaml"/>
    <param name="log_output_path" type="str" value="$(find kimera_distributed)/logs/basestation/"/>
  </node> 

  <!-- Pose Graph Visualizer -->
  <include file="$(find pose_graph_tools)/launch/posegraph_view.launch" >
    <arg name="frame_id"      value="world" />
    <arg name="graph_topic"   value="pose_graph_optimized" />
    <arg name="ns"            value="kimera_centralized"/>
  </include>

  <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find kimera_distributed)/rviz/kimera_centralized.rviz"></node>

</launch>